app/src/marketplace/containers/EditProductPage/index.jsx

```diff
@@ -144,8 +145,9 @@ const EditProductPage = ({ product }: { product: Product }) => {
 
     // clear data union secrets when unmounting
     useEffect(() => () => {
+        resetDataUnion()
```

Comment:
Reset active DU id in `dataUnion` state when leaving editor... If `beneficiaryAddress` is not set on a product, DU info is not loaded when entering -> leaves old active DU in state